From 8a170f21338e8427835d6e265820963208017af1 Mon Sep 17 00:00:00 2001
From: Aaradhana Sahu <quic_aarasahu@quicinc.com>
Date: Tue, 23 Aug 2022 16:13:08 +0530
Subject: [PATCH] ath12k: fix antenna parameter is zero in iw phyx

Available antenna parameter Tx and Rx is zero and configured antenna
is missing in iw phyx info.

Fixing by filling cap variable in ath12k_mac_hw_register function

Signed-off-by: Aaradhana Sahu <quic_aarasahu@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index 5ed8bc5..448aba1 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -10620,6 +10620,7 @@ static int ath12k_mac_hw_register(struct ath12k_hw *ah)
 	for (i = 0; i < ah->num_radio; i++) {
 		ab = ar->ab;
 		pdev = ar->pdev;
+		cap = &ar->pdev->cap;
 		if (ab->pdevs_macaddr_valid) {
 			ether_addr_copy(ar->mac_addr, pdev->mac_addr);
 		} else {
-- 
2.17.1

